### DISCUSSION TITLE
setValueProvider not working as before

### DISCUSSION THREAD

**Discussion by ana-bonnet** [airbnb/lottie-ios#1846] [General]
Hello - hope you are all good.

We just update the SDK from 3.4.3 to 4.0.1 using cocoa pods. We are using a few simple animations that contains only one color, for example we have a circular loader and we have been updating the color with the code below, have been working great but since we updated the library it doesn't work anymore, the JSON hasn't change, so does any of this flow have been updated in the 4.0.1 version? Do I need to change anything?

`let animationView: LottieAnimationView = LottieAnimationView(name: "loader")`
`let keypath = AnimationKeypath(keys: ["**", "Stroke 1", "**", "Color"])`
`let colorProvider = ColorValueProvider(theme.tintColor.lottieColorValue)`  
`animationView.setValueProvider(colorProvider, keypath: keypath)`

Thanks a lot in advance.

**Comment by calda** [comment]
Thanks for the report. I created an issue for this at https://github.com/airbnb/lottie-ios/issues/1847. Could you share a copy of your animation json over in that issue ticket? Thanks!

**Reply by calda** [reply]
As a workaround until we fix this issue you can switch back to using the main thread rendering engine -- just pass `configuration: LottieConfiguration(renderingEngine: .mainThread)` to your `LottieAnimationView`.

**Reply by ana-bonnet** [reply]
Thank you so much, just added the animation copy in the issue! Will try that solution in the meantime

**Reply by gee224** [reply]
Facing the same issue. Work around mentioned by @calda is not working for me. @ana-bonnet did you get around this issue?

**Comment by ana-bonnet** [comment]
Hey 👋 not yet, I move back to a version where did work in the meantime,
but try out what you mention above and didn’t work either.

Will let you know if I found something


On Tue, Jan 10, 2023 at 12:51 AM, gee224 ***@***.***> wrote:

> Facing the same issue. Work around mentioned by @calda
> <https://github.com/calda> is not working for me. @ana-bonnet
> <https://github.com/ana-bonnet> did you get around this issue?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/airbnb/lottie-ios/discussions/1846#discussioncomment-4639757>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ATR45YLYTXSQJZ24M3F3N7DWRSXABANCNFSM6AAAAAAS4A2OEE>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>
-- 
[image: twitter] <https://twitter.com/bonnetelectric>
[image: linkedin] <https://www.linkedin.com/company/bonnet/>
[image: instagram] <https://www.instagram.com/bonnetelectric/>
Ana Isabel Marquez

iOS Lead Developer

Bonnet
***@***.***
https://www.joinbonnet.com/om/
